getting-started.ipynb

```diff
@@ -0,0 +1,601 @@
+        "from torchvision.datasets import MNIST\n",
+        "from torchvision.models import resnet18\n",
+        "from torchvision.transforms import Compose, Normalize, ToTensor\n",
+        "from tqdm import tqdm\n",
+        "\n",
+        "from ignite.engine import Engine, Events, create_supervised_trainer, create_supervised_evaluator\n",
+        "from ignite.metrics import Accuracy, Loss"
+      ],
+      "outputs": [],
+      "metadata": {
+        "id": "Saizk3heEcFz"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Speed things up by setting [device](https://pytorch.org/docs/stable/tensor_attributes.html#torch.torch.device) to `cuda` if available else `cpu`."
+      ],
+      "metadata": {
+        "id": "ecMYtJF7OvgT"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 8,
+      "source": [
+        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")"
+      ],
+      "outputs": [],
+      "metadata": {
+        "id": "sdjDKcFhOuQn"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Define a class of your model or use the predefined ResNet18 model (modified for MNIST) below, instantiate it and move it to device:"
+      ],
+      "metadata": {
+        "id": "4r_PUH1yEcF1"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "source": [
+        "class Net(nn.Module):\n",
+        "\n",
+        "    def __init__(self, in_channels=1):\n",
+        "        super(Net, self).__init__()\n",
+        "        \n",
+        "        # Changed the output layer to output 10 classes instead of 1000 classes\n",
+        "        self.model = resnet18(num_classes=10)\n",
+        "\n",
+        "        # Changed the input layer to take grayscale images for MNIST instaed of RGB images\n",
+        "        self.model.conv1 = nn.Conv2d(1, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)\n",
+        "\n",
+        "\n",
+        "    def forward(self, x):\n",
+        "        return self.model(x)\n",
+        "\n",
+        "model = Net().to(device)"
+      ],
+      "outputs": [],
+      "metadata": {
+        "id": "dVSVAT0OEcF1"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Now let us define the training and validation datasets (as\n",
+        "[torch.utils.data.DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader))\n",
+        "and store them in `train_loader` and `val_loader` respectively. We have\n",
+        "used the [MNIST](https://pytorch.org/vision/stable/datasets.html#mnist)\n",
+        "dataset for ease of understanding.\n"
+      ],
+      "metadata": {
+        "id": "DDIW2zedEcF3"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "source": [
+        "def get_data_loaders(train_batch_size, val_batch_size):\n",
```

Comment:
Why introduce this function rather than the plain code ?

Reply:
This function was used in several MNIST ignite/examples. Although I can replace it with plain code!

Reply:
Done!